File: BeagleBone/Black/input.js
#!/usr/bin/env node
////////////////////////////////////////
//	input.js
//      Responds to changes on P8_19 via a callback.
//	Wiring:	Connect P8_19 to P9_14 with a 1kOhm resistor.
//	Setup:	
//	See:	
////////////////////////////////////////
const b = require('bonescript');

const outputPin = "P9_14";
const inputPin  = "P8_19";
const ledPin    = "USR3";
const mydelay = 250;
var state   = b.LOW;

console.log('Please connect ' + inputPin + ' to ' + outputPin +
    ' with a 1kOhm resistor');
b.pinMode(inputPin, b.INPUT);
b.pinMode(outputPin, b.OUTPUT);
b.pinMode(ledPin, b.OUTPUT);

b.digitalWrite(outputPin, b.LOW);
b.attachInterrupt(inputPin, setLED, b.CHANGE);

toggle();

function setLED(x) {
    b.digitalWrite(ledPin, x.value);
}

function toggle() {
    state = (state == b.LOW) ? b.HIGH : b.LOW;
    b.digitalWrite(outputPin, state);
    setTimeout(toggle, mydelay);        // Waut a bit, and then toggle again
}

File: BeagleBone/Black/input.py
#!/usr/bin/env python3
# ////////////////////////////////////////
# //	input.py
# //      Responds to changes on P8_19 via a callback.
# //	Wiring:	Connect a switch between inputPin and 3.3V
# //            Attach and LED through a 220 Ohm resistor to outputPin
# //	Setup:	
# //	See:	
# ////////////////////////////////////////
import Adafruit_BBIO.GPIO as GPIO
import time

outputPin = "P9_14";
inputPin  = "P8_14";
ledPin    = "USR3";
state   = 0;

# Set the direction on the pins
GPIO.setup(outputPin, GPIO.OUT)
GPIO.setup(ledPin,    GPIO.OUT)
GPIO.setup(inputPin,  GPIO.IN)

# This calback is called everytime the input changes
# channel is the name of the pin that changed
def my_callback(channel):
    # Read the current value of the input
    state = GPIO.input(channel)
    print('Edge detected on channel {}, value={}'.format(channel, state))
    # Write it out
    GPIO.output(outputPin, state)
    GPIO.output(ledPin, state)

# This is a non-blocking event 
GPIO.add_event_detect(inputPin, GPIO.BOTH, callback=my_callback) 

try:
    while True:             # Do something while waiting for event
        time.sleep(100)

except KeyboardInterrupt:
    print("Cleaning up...")
    GPIO.cleanup()
